cpp/include/raft/linalg/detail/map_then_reduce.cuh

```diff
@@ -86,8 +88,8 @@ void mapThenReduceImpl(OutType* out,
                        Args... args)
 {
   raft::update_device(out, &neutral, 1, stream);
-  const int nblks = raft::ceildiv(len, (size_t)TPB);
-  mapThenReduceKernel<InType, OutType, MapOp, ReduceLambda, TPB, Args...>
+  const int nblks = raft::ceildiv(len, (IdxType)TPB);
```

Comment:
Would you consider for new code using C++ casts (`IdxType(TPB)` or `static_cast<IdxType>(TPB)`) instead of C-style casts?